### DISCUSSION TITLE
Temp/Install.bat

### DISCUSSION THREAD

**Discussion by zedd-0** [rustdesk/rustdesk#10392] [General]
### Bug Description

Hello,

I noticed the following command in the "RustDesk_install.bat":
"reg delete HKEY_CLASSES_ROOT\.rustdesk /f"

This is problematic for me, especially for (silent) updates via "Windows Winget".
In this registry key, the values "DESKTOPSHORTCUTS" and "STARTMENUSHORTCUTS" are stored.

I had both set to "0" to avoid creating shortcuts.
However, since this registry key is deleted by the command above,
my preferences are ignored during an update.

I hope the installer will be made more update-friendly in the future.

Personally, I think that "HKEY_CLASSES_ROOT\.rustdesk"
is not a suitable place to store application settings.
Microsoft recommends the following registry paths for such settings:

"HKCU\Software\<APP>" for user-specific settings
"HKLM\SOFTWARE\<APP>" for system-wide settings

It would be a good idea to consider these guidelines
and adjust the storage location for settings accordingly.

I also noticed several "RustDesk_*_shortcut.vbs" files in the Temp directory.
Since Microsoft has discontinued support for VBS,
I would recommend migrating to PowerShell in a timely manner.
In PowerShell, "WScript.Shell" can also be used.

Thanks a lot.

### How to Reproduce

-

### Expected Behavior

-

### Operating system(s) on local (controlling) side and remote (controlled) side

Win 11 24h2

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.8

### Screenshots

-

**Comment by rustdesk** [comment]
pr is welcome

**Reply by zedd-0** [reply]
version v1.3.9
running "winget upgrade rustdesk"

my settings in the registry:
```
[HKEY_CLASSES_ROOT\.rustdesk]
@=""
"DESKTOPSHORTCUTS"="0"
"STARTMENUSHORTCUTS"="1"
``` 

...thy was overwritten again.
```
[HKEY_CLASSES_ROOT\.rustdesk]
@=""
"DESKTOPSHORTCUTS"="1"
"STARTMENUSHORTCUTS"="1"
"PRINTER"="1"
```

I still believe that an entry for the file extension 'HKEY_CLASSES_ROOT.rustdesk' is a poorly chosen place for general configurations. Microsoft provides 'HKEY_CURRENT_USER\Software' or 'HKEY_LOCAL_MACHINE\Software' for this purpose.

I also do not understand why my 'Issue' is being turned into a 'Discussion'.

I want my system to be as clean as possible, which is why I see the non-compliance with recommendations/guidelines as critical.
In any case, these settings should not be so easily deletable/overwritable. A separate registry branch is appropriate here.


thanks.

**Comment by zedd-0** [comment]
what is "PR"?
if "PR" ist "Pull Requests" (i asked chatgpt)

sadly, i am not a programmer, just an end user. 
I my best tried to describe the problem as best as I could and indicate locations.
In the hope that the application will be improved as a result.
sry.
